stellarphot/photometry/photometry.py

```diff
@@ -1,138 +1,408 @@
+                    "to use RA/Dec ... SKIPPING THIS IMAGE!")
 
-    # Extract x,y coordinates from sources table, construct aperture and
-    # annulus objects from coordinates, and perform aperture photometry
-    coords = np.array([sources['xcentroid'], sources['ycentroid']]).T
-    apertures = CircularAperture(coords, aperture_radius)
-    annulus = CircularAnnulus(coords, inner_annulus, outer_annulus)
-    phot_table = aperture_photometry(ccd_image, apertures)
-    phot_table_1 = aperture_photometry(ccd_image, annulus)
+    # Compute RA/Dec if not already provided
+    if not sourcelist.has_ra_dec:
+        try:
+            skypos = ccd_image.wcs.pixel_to_world(xs, ys)
+            ra = skypos.ra.value
+            dec = skypos.dec.value
+        except AttributeError:
+            ra = [np.nan] * len(xs)
+            dec = [np.nan] * len(ys)
+
+    # Define apertures and annuli for the aperture photometry
+    aper_locs = np.array([xs, ys]).T
+    apers = CircularAperture(aper_locs, r=aperture_radius)
+    anuls = CircularAnnulus(aper_locs, r_in=inner_annulus, r_out=outer_annulus)
+
+    # Perform the aperture photometry
+    photom = aperture_photometry(ccd_image.data, (apers, anuls),
+                                mask=ccd_image.mask, method='center')
+
+    # Add source ids to the photometry table
+    photom['star_id'] = star_ids
+    photom['ra'] = ra
+    photom['ra'].unit = u.deg
+    photom['dec'] = dec
+    photom['dec'].unit = u.deg
+
+    # Drop ID column from aperture_photometry()
+    del photom['id']
+
+    # Add various CCD image parameters to the photometry table
+    if fname is not None:
+        photom['file'] = fname
+    else:
+        photom['file'] = [''] * len(photom)
+
+    # Search for exposure keyword in header and set exposure if found
+    exp_kw = ["EXPOSURE", "EXPTIME", "TELAPSE", "ELAPTIME", "ONTIME", "LIVETIME"]
+    matched_kw = None
+    for kw in exp_kw:
+        if kw in ccd_image.header:
+            matched_kw = kw
+            break
+
+    if matched_kw is None:
+        raise ValueError("single_image_photometry: None of the allowed exposure "
+                         f"keywords ({format(', '.join(exp_kw))}) found in the header "
+                         "... SKIPPING THIS IMAGE!")
+    else:
+        photom['exposure'] = [ccd_image.header[matched_kw]] * len(photom) * u.second
 
-    # Obtain the local background/pixel and net flux between the aperture and
-    # annulus objects
-    n_pix_ap = apertures.area
-    n_pix_ann = annulus.area
+    try:
+        photom['date-obs'] = Column(data=Time([ccd_image.header['DATE-OBS']]
+                                              * len(photom),
+                                    format='isot', scale='utc'), name='date-obs')
+    except KeyError:
+        raise ValueError("single_image_photometry: 'DATE-OBS' not found in CCD image "
+                         "header ... SKIPPING THIS IMAGE!")
```

Comment:
Could these checks that result in skipping the image be move earlier to save a bit of time? No reason to run `aperture_photometry` if we are going to skip it.

Reply:
They were done here because I was adding them to the photom table which is not created until aperture_photometry is run.  I will fix this in the multi_image_photometry() version of photometry.py.